### PR TITLE
Dedupe issues, handle non-existent issues

### PR DESCRIPTION
**Pull request by ZStriker19**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -99,33 +104,49 @@ def get_closed_issues(
...
+                else:
+                    if gh_issue.state == "closed" and gh_issue not in [ghi[1] for ghi in closed_issues]:
+                        closed_issues.append((issue, gh_issue))
     return closed_issues
 
 
+def get_unique_issues(
+    issues: List[FileIssue]
+) -> Dict[str, List[FileIssue]]:
+    """Return all the name and line number of all the files that contain the issue"""
```
Comment (Kyle-Verhoog) [inline main.py]: This doc string needs to be updated

File: main.py [hunk 2]
```diff
@@ -99,33 +104,49 @@ def get_closed_issues(
...
+                dedupe_issues[ref].append(issue)
+                break
+        else:
+            dedupe_issues[issue.ref] = [issue]
+    return dedupe_issues
+
+
 def get_issue_locations(
-    issues: List[FileIssue], issue: FileIssue
+    unique_issues: Dict[str, List[FileIssue]], closed_issue: FileIssue
```
Comment (Kyle-Verhoog) [inline main.py]:
```suggestion
    issues: Dict[str, List[FileIssue]], issue: FileIssue
```

can revert these changes

File: main.py [hunk 3]
```diff
@@ -160,16 +183,23 @@ def get_issue_locations(
...
                 )
-                break
-        else:
-            gh_repo.create_issue(
-                title=f"Upstream issue {issue.ref} closed",
-                body=body,
-                labels=LABELS,
-            )
+            else:
+                logging.info(f"Would create issue in repo {GH_REPO}:\nUpstream issue {issue.ref} {GH_REPO}\n\n {body}"),
```
Comment (Kyle-Verhoog) [inline main.py]:
```suggestion
                logging.info(f"Would create issue in repo '{GH_REPO}':\nUpstream issue {issue.ref}\n\n{body}"),
```

File: main.py [hunk 4]
```diff
@@ -160,16 +183,23 @@ def get_issue_locations(
...
+            if not DRY_RUN:
+                if issue.ref in repo_issue.title or issue.ref in repo_issue.body:
+                    # repo issue already exists for the upstream issue, update it
+                    # in case any references have been removed.
+                    repo_issue.edit(
+                        body=body,
+                    )
+                    break
+            else:
+                logging.info(f"Would edit issue number {repo_issue.number} in repo {GH_REPO}:\nUpstream issue {issue.ref}\n\n {body}"),
```
Comment (Kyle-Verhoog) [inline main.py]:
```suggestion
                logging.info(f"Would edit issue number {repo_issue.number} in repo {GH_REPO}:\nUpstream issue {issue.ref}\n\n{body}"),
```

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -9,7 +9,9 @@ jobs:
     - uses: actions/setup-python@v4
       with:
         python-version: "3.10"
+    - name: Pass Inputs to Shell
```
Comment (Kyle-Verhoog) [inline .github/workflows/test.yml]: revert this

File: main.py [hunk 6]
```diff
@@ -160,16 +183,23 @@ def get_issue_locations(
...
+            if not DRY_RUN:
+                if issue.ref in repo_issue.title or issue.ref in repo_issue.body:
+                    # repo issue already exists for the upstream issue, update it
+                    # in case any references have been removed.
+                    repo_issue.edit(
+                        body=body,
+                    )
+                    break
+            else:
+                logging.info(f"Would edit issue number {repo_issue.number} in repo {GH_REPO}:\nUpstream issue {issue.ref}\n\n{body}"),
```
Comment (Kyle-Verhoog) [inline main.py]:
```suggestion
                logging.info(f"Would edit issue number {repo_issue.number} in repo '{GH_REPO}':\nUpstream issue {issue.ref}\n\n{body}"),
```

Comment (ZStriker19) [inline main.py]:
```suggestion
                logging.info(f"Would edit issue number {repo_issue.number} in repo `{GH_REPO}`:\nUpstream issue {issue.ref}\n\n{body}"),
```

**Review by Kyle-Verhoog — APPROVED** [review]
